Logging operator CrashLoopBackOff with Panic Error Message

**Describe the bug**:
Logging operator reports CrashLoopBackOff with Panic Error Message

**Expected behaviour**:
Logging operator UP and Running

**Steps to reproduce the bug**:
Follow the exact steps of readme for logging operator.


**Environment details**:
- Kubernetes version (e.g. v1.15.2): 1.21
- Cloud-provider/provisioner (e.g. AKS, GKE, EKS, PKE etc): EKS
- logging-operator version (e.g. 2.1.1):  3.15.0
- Install method (e.g. helm or static manifests): helm
- Logs from the misbehaving component (and any other relevant logs):

{"level":"info","ts":1637060223.3517358,"logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":":8080"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1625411]

goroutine 1 [running]:
github.com/banzaicloud/logging-operator/pkg/k8sutil.(*Cached).RESTMapping(0xc0001c21e0, 0x0, 0x0, 0x168561b, 0x4, 0xc00037eb00, 0x1, 0x1, 0xc0001eb260, 0x1bb7820, ...)
        /workspace/pkg/k8sutil/cache.go:112 +0x31
sigs.k8s.io/controller-runtime/pkg/client.(*clientCache).newResource(0xc000198d80, 0x0, 0x0, 0x19b9daf, 0x2, 0x168561b, 0x4, 0x1680001, 0x8, 0x0, ...)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/client/client_cache.go:66 +0x273
sigs.k8s.io/controller-runtime/pkg/client.(*clientCache).getResource(0xc000198d80, 0x1bbfbe0, 0xc000326000, 0x0, 0x0, 0x0)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/client/client_cache.go:102 +0x286
sigs.k8s.io/controller-runtime/pkg/client.(*typedClient).List(0xc000326cb0, 0x1bea008, 0xc0000a8000, 0x1bfda20, 0xc000326000, 0xc00037ea40, 0x1, 0x1, 0x7f3594b33e40, 0x0)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/client/typed_client.go:148 +0x94
sigs.k8s.io/controller-runtime/pkg/client.(*client).List(0xc000326cb0, 0x1bea008, 0xc0000a8000, 0x1bfda20, 0xc000326000, 0xc00037ea40, 0x1, 0x1, 0x0, 0x0)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/client/client.go:287 +0x1b3
main.detectContainerRuntime(0x1bea008, 0xc0000a8000, 0x7f3594891f28, 0xc000326cb0, 0x1bf31b0, 0xc0002f3060)
        /workspace/main.go:164 +0xc6
main.main()
        /workspace/main.go:131 +0x8d3


/kind bug

--- a/pkg/k8sutil/cache.go
+++ b/pkg/k8sutil/cache.go
@@ -34,7 +34,9 @@ func NewCached(config *rest.Config) (meta.RESTMapper, error) {
 			return apiutil.NewDiscoveryRESTMapper(config)
 		},
 	}
-	c.flush()
+	if err := c.flush(); err != nil {
+		return nil, err
+	}
 	return c, nil
 }
 

